### PR TITLE
fix issue 1778

### LINKED ISSUE #1778: [BUG] Kompose does not mount secrets correctly
### Expected Behavior

secrets defined in compose.yaml should be mounted in the pods in `/run/secrets/<secret>`

### Actual Behavior

`/run/secrets/<secret>` is mounted as a directory

### Steps To Reproduce

1. Add secrets to compose.yaml

```
services:
  myservice:
    ...
    secrets:
      - config-ini

secrets:
  config-ini:
    file: secrets/Config.ini
```

2. kompose convert

3. examine generated pod config file:

The result is:
```
volumeMounts:
  - mountPath: /run/secrets/config-ini
     name: config-ini
```

Which appears correct at a glance, but when running, I get the error that `config-ini` is a directory.

### Kompose Version

```Text
1.31.2 (HEAD)
```

### Docker-Compose file

_No response_

### Anything else?

I was able to fix the pod file by adding subPath:
```
volumeMounts:
  - mountPath: /run/secrets/config-ini
    subPath: config-ini
    name: config-ini
```

With subPath, it gets mounted as a file instead of a directory.

I don't know k8 very well, I would guess it's mounted as a directory because secrets can hold more than one key-value pair. In that case it wouldn't necessarily make sense to put all the key-value pairs into one file, especially if each key should be its own file.

But with docker compose, secrets are only one key->value, so that behavior should be translated.

### PR DESCRIPTION
**Pull request by sosan**
#### What type of PR is this?
/kind bug
#### Which issue(s) this PR fixes:
* fix issue https://github.com/kubernetes/kompose/issues/1778 
* add tests with hardcoded values to mokup file access with a file that exist in repo

Fixes #1778

#### Special notes for your reviewer:
i runned action Kompose CI, but get error from action, i checked result and expect and it's very similar 
https://github.com/sosan/kompose/actions/runs/7479241247/job/20355974505
```
convert::expect_success: 
Running: 'kompose -f /home/runner/work/kompose/kompose/script/test/fixtures/configmap-pod/docker-compose.yml convert --stdout --with-kompose-annotation=false' expected_output: '/home/runner/work/kompose/kompose/script/test/fixtures/configmap-pod/output-k8s.yaml'_ __ __ _| |_ _ / _|/ _| between /home/runner/work/kompose/kompose/script/test/fixtures/configmap-pod/output-k8s.yaml, 
four documents / _' | | | | |_| |_ and /tmp/kompose/test-stdout, 
four documents | (_| | |_| | _| _| \__,_|\__, |_| |_| returned five differences |___/ metadata (Service/default/redis) 

one map entry removed: creationTimestamp: null metadata (Pod/default/redis) - 
one map entry removed: creationTimestamp: null spec.containers.redis (Pod/default/redis) - 
one map entry removed: resources: {} metadata (ConfigMap/default/foo-env) - 
one map entry removed: creationTimestamp: null metadata (ConfigMap/default/bar-env) - 
one map entry removed: creationTimestamp: null
FAIL: converted output does not match
```

### CODE REVIEW

**Comment by cdrage** [comment]
> /hold

I forget why this is on hold haha

**Comment by AhmedGrati** [comment]
@cdrage e2e tests were failing, and I opened a PR to fix them. Can you please rebase @sosan?

**Comment by AhmedGrati** [comment]
ping @sosan

**Comment by sosan** [comment]
@AhmedGrati yup

**Comment by AhmedGrati** [comment]
@sosan can you rebase please?

**Comment by sosan** [comment]
@AhmedGrati done

File: pkg/transformer/kubernetes/kubernetes_test.go [hunk 1]
```diff
@@ -1145,3 +1145,68 @@ func TestNamespaceGenerationBlank(t *testing.T) {
...
+	}
+
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    []*api.Secret
+		wantErr bool
+	}{
+		// TODO: Add test cases.
```
Comment (cdrage) [inline pkg/transformer/kubernetes/kubernetes_test.go:1175]: Im assuming we can remove this?

**Review by cdrage — CHANGES_REQUESTED** [review]

File: pkg/transformer/kubernetes/kubernetes_test.go (hunk 1 above)
Comment (sosan) [reply pkg/transformer/kubernetes/kubernetes_test.go]: added one more test to generate secrets from file and merged new additions from main

**Review by cdrage — APPROVED** [review]
Much better and thank you for adding the tests. This LGTM!

**Pull request merged**